src/coreclr/vm/comdelegate.cpp

```diff
@@ -189,16 +189,57 @@ class ShuffleIterator
         if (m_currentByteStackIndex < m_argLocDesc->m_byteStackSize)
         {
             const unsigned byteIndex = m_argLocDesc->m_byteStackIndex + m_currentByteStackIndex;
-            index = byteIndex / TARGET_POINTER_SIZE;
+
+#if !defined(TARGET_OSX) || !defined(TARGET_ARM64)
             m_currentByteStackIndex += TARGET_POINTER_SIZE;
 
             // Delegates cannot handle overly large argument stacks due to shuffle entry encoding limitations.
+            index = byteIndex / TARGET_POINTER_SIZE;
             if (index >= ShuffleEntry::REGMASK)
             {
                 COMPlusThrow(kNotSupportedException);
             }
 
-            return -(int)byteIndex;
```

Comment:
Can you please fix the function description comment? This change makes it invalid.